src/librbd/internal.cc

```diff
@@ -2150,7 +2150,8 @@ bool compare_by_name(const child_info_t& c1, const child_info_t& c2)
 		       int (*cb)(uint64_t, size_t, const char *, void *),
 		       void *arg)
   {
-    utime_t start_time, elapsed;
+    mono_time start_time;
```

Comment:
Question: I assume there was discussion about using std::steady_clock?

Reply:
Well, we want to use steady_clock whenever we're measuring the passage of time, as it' notsubject to changes to the system's clock. We want to move away from `ceph_clock_now()`, which uses the system clock, whenever we subtract timestamps.

Reply:
@adamemerson I'm not fully convinced that the standard clocks are inadequate here, but knowing that it was considered and that your choice was a reasoned one does make me feel better. OTOH, glock_gettime() is /also/ implementation dependent with respect to its resolution, so I'm not really sure what you gain.

Reply:
While `clock_gettime(2)` is implementation dependent, we do know pretty well what the specific flags do on various systems (I've actually spent some time researching the differences in call-time-overhead on Linux) so under the hood our mono_clock uses CLOCK_MONOTONIC on Linux and CLOCK_MONOTONIC_PRECISE on FreeBSD, and coarse_mono_clock uses CLOCK_MONOTONIC_COARSE on Linux and CLOCK_MONOTONIC_FAST on FreeBSD. The big thing rolling our clocks gives us is that for cases where we want a steady clock for internal instrumentation (performance counters, printing logs, etc.) we don't end up paying the exorbitant overhead that we incur when we measure precise delays, say.

Reply:
And believe me, the difference in overhead is surprisingly large.

Reply:
Do you have the results anywhere? I'd be curious to see the overhead. FWIW, I've spent quite some time understanding the internals of `clock_gettime()` as well, although not from an overhead perspective.

Reply:
+1 sounds like this is well researched! Thanks for the explanation. Curious also about @mogeb's question.

Reply:
FWIW, libcxx appears to be currently using this for steady_clock if a monotonic clock is available:
    if (0 != clock_gettime(CLOCK_MONOTONIC, &tp))

Reply:
@mogeb 
It is a rather well known fact that there is a substantial difference in overhead between the Linux and FreeBSD implementations. Just about any smaller performance comparison trips over that, artificially showing Linux delivering better performance.
Cost of the calls on FreeBSD are dependent on the type of accuracy required.
Which is why I was happy with the changes that went in Ceph, to only use accurate clocks when really needed.

So I would appreciate careful treading in this area. And if possible only change it if really needed.
And perhaps benchmark on both FreeBSD and Linux.
Give me some unit_tests: old versus new, and I'll try and run them for you on FreeBSD

Reply:
@mogeb 
Also see:
    https://www.freebsd.org/cgi/man.cgi?query=clock_gettime
where it specifies the fact that the `PRECISE` extension comes at a cost.

Reply:
@wjwithagen good point, understood. I've reworked the PR to use the coarse steady clock when better fit. I'll add some benchmarks as well.